doc/api/test_runner.md

```diff
@@ -0,0 +1,275 @@
+calling the test context's `skip()` method. Both of these options support
+including a message that is displayed in the TAP output as shown in the
+following example.
+
+```js
+// The skip option is used, but no message is provided.
+test('skip option', { skip: true }, (t) => {
+  // This code is never executed.
+});
+
+// The skip option is used, and a message is provided.
+test('skip option with message', { skip: 'this is skipped' }, (t) => {
+  // This code is never executed.
+});
+
+test('skip() method', (t) => {
+  // Make sure to return here as well if the test contains additional logic.
+  t.skip();
+});
+
+test('skip() method with message', (t) => {
+  // Make sure to return here as well if the test contains additional logic.
+  t.skip('this is skipped');
+});
+```
+
+## Extraneous asynchronous activity
+
+Once a test function finishes executing, the TAP results are output as quickly
+as possible while maintaining the order of the tests. However, it is possible
+for the test function to generate asynchronous activity that outlives the test
+itself. The test runner handles this type of activity, but does not delay the
+reporting of test results in order to accommodate it.
+
+In the following example, a test completes with two `setImmediate()`
+operations still outstanding. The first `setImmediate()` attempts to create a
+new subtest. Because the parent test has already finished and output its
+results, the new subtest is immediately marked as failed, and reported in the
+top level of the file's TAP output.
+
+The second `setImmediate()` creates an `uncaughtException` event.
+`uncaughtException` and `unhandledRejection` events originating from a completed
+test are handled by the `test_runner` module and reported as diagnostic
+warnings in the top level of the file's TAP output.
+
+```js
+test('a test that creates asynchronous activity', (t) => {
+  setImmediate(() => {
+    t.test('subtest that is created too late', (t) => {
+      throw new Error('error1');
+    });
+  });
+
+  setImmediate(() => {
+    throw new Error('error2');
+  });
+
+  // The test finishes after this line.
+});
+```
+
+## `test([name][, options][, fn])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `name` {string} The name of the test, which is displayed when reporting test
+  results. **Default:** The `name` property of `fn`, or `'<anonymous>'` if `fn`
+  does not have a name.
+* `options` {Object} Configuration options for the test. The following
+  properties are supported:
+  * `concurrency` {number} The number of tests that can be run at the same time.
+    If unspecified, subtests inherit this value from their parent.
+    **Default:** `1`.
+  * `skip` {boolean|string} If truthy, the test is skipped. If a string is
+    provided, that string is displayed in the test results as the reason for
+    skipping the test. **Default:** `false`.
+* `fn` {Function} The function under test. The test fails if this function
```

Comment:
To clarify that async functions are OK here. Feel free to ignore as there are already examples showing their use so it may not be necessary.
```suggestion
* `fn` {Function|AsyncFunction} The function under test. The test fails if this function
```